client/src/controllers/ActionController.test.js

```diff
@@ -29,4 +29,37 @@ describe('ActionController', () => {
     expect(new FormData(form).get('csrfmiddlewaretoken')).toBe('potato');
     expect(new FormData(form).get('next')).toBe('http://localhost/');
   });
+
+  describe('click', () => {
+    let app;
+    beforeEach(() => {
+      document.body.innerHTML = `
+      <button
+      type="button"
+        id="button"
+        data-controller="w-action"
+        data-action="some-event->w-action#click"
+      >
+        Foo
+      </button>
+      `;
+      app = Application.start();
+      app.register('w-action', ActionController);
+    });
+
+    afterEach(() => {
+      app.stop();
+    });
+
+    it('should call click method when button is clicked via Stimulus action', () => {
+      const btn = document.getElementById('button');
+      const clickMock = jest.fn();
+      btn.addEventListener('some-event', clickMock);
+
+      const event = new CustomEvent('some-event');
+      btn.dispatchEvent(event);
+      
+      expect(clickMock).toHaveBeenCalled();
```

Comment:
Hey. Thanks again. This is quite close but still missing one key part. 

Let's walk through what your test does

1. Create some HTML with a Stimulus controlled element with a Stimulus action that listens for 'some-event' that will call the 'click' method.
2. Set up a new Stimulus application and register the ActionController
3. In the unit test, create a mock function, attach this mock as the call back for a 'some-event' listener
4. Trigger the custom event 'some-event' on the button
5. Check the callback was called. 

The problem here is your test is saying, when I dispatch the custom event, check the custom event is dispatched. Nothing here tests the button itself was clicked. 

Don't rewrite the whole thing, this just needs the clickMock function to be set up differently. 

Instead of 
```
btn.addEventListener('some-event', clickMock);
```

Replace this line with something similar to an earlier version of.the PR, where you use the mock function to replace the button's click function. 

```
btn.click , clickMoc);
```

Then, you'll be testing that dispatching the custom event 'some-event', calls the method 'click' on the controller (via stimulus action data attributes). Which in turn should call the buttons click function, which you have replaced with your mock.